Student results teacher view

Tomatsuppe11211 opened the pull request:
### What does this PR do?

Added student results (teachers view) with HTML and CSS



### Description of Task to be completed?

Remade HTML coding and adding tailwind CSS with v4 update.



### How should this be manually tested?

Check out the page in live preview to check styling, functions and scaling.


### Any background context you want to provide?
There are yet somethin on the to do list here, so this is just for updating the branch.

Tomatsuppe11211 committed:
Adding student results page (Teacher's view)

I have added the student results page (teachers view).

I have created the file localy on my computer while there was problems with tailwind css updates. THis page is running good in this branch and on my computer, but thes is some visually issues that came up when i added the code to this branch.

Tomatsuppe11211 committed:
Removing sidebar

I have removed the sidebar as i got to know that we were not supposed to have it in our coed yet.

Tomatsuppe11211 committed:
Merge branch 'develop' into student-results-teacher.view

kristinatonnessen commented on templates/student-result-teacher-view.html:
```diff
@@ -0,0 +1,196 @@
+    <title>Edugate | teacher: edit students results</title>
+    <script src="https://kit.fontawesome.com/8908c40066.js" crossorigin="anonymous"></script>
+    <link rel="stylesheet" href="../dist/output.css">
+    <link rel="stylesheet" href="../css/styles.css">
```
this can be removed

Tomatsuppe11211 replied:
Removed line 10

i comment just to use this as checkbox XD

kristinatonnessen commented on templates/student-result-teacher-view.html:
```diff
@@ -0,0 +1,196 @@
+
+
+            <div class="flex flex-row mt-5 md:mt-8 items-center justify-center text-center space-x-2 lg:space-x-4 xl:space-x-6">
+                <i class="fa-solid fa-angles-left border-1 border-gray-300 p-1 md:p-2 w-5 md:w-8"></i>
```
rounded borders and sizing the borders in mobile view
![Skjermbilde 2025-01-31 kl  23 41 59](https://github.com/user-attachments/assets/7370b929-c9dc-4440-b14b-86b3fee0bbb4)

Tomatsuppe11211 replied:
Added roudn borders:

![image](https://github.com/user-attachments/assets/4cddd2e3-6334-4754-85d5-ea50150bc975)

kristinatonnessen commented on templates/student-result-teacher-view.html:
```diff
@@ -0,0 +1,196 @@
+            </div>
+
+            <div class="flex justify-center mt-5 md:mt-8">
+                <button class="mt-5 border-2 border-[#2196F3] p-1 w-35 md:w-45 text-[#2196F3] hover:bg-[#BBDEFB]">Back to Dashboard</button>
```
update to the custom class secondary-button

Tomatsuppe11211 replied:
Removed content and added class secondary button

![image](https://github.com/user-attachments/assets/1a250d57-a453-4f0b-b24d-4e270cff028d)

I will work on scaling

kristinatonnessen commented on templates/student-result-teacher-view.html:
```diff
@@ -0,0 +1,196 @@
+                                <a href="#"><p class="text-[#4CAF50] hidden md:block">Edit</p></a>
+                            </td>
+                            <td class="hidden md:block w-15">
+                                <button class="p-1 lg:p-2 rounded-sm bg-[#FF5722] text-white hover:bg-[#E64A19]">Delete</button>
```
change all delete buttons to custom class alert-button

Tomatsuppe11211 replied:
![image](https://github.com/user-attachments/assets/813eb8ee-7e8e-4d4c-977a-dcdcf1ea1260)

Done. Will work on scaling. They look to big now
![image](https://github.com/user-attachments/assets/927dc384-eac4-4019-b28f-64365900b5b3)

kristinatonnessen commented on templates/student-result-teacher-view.html:
```diff
@@ -0,0 +1,196 @@
+    <header></header>
+    <main class="flex flex-row justify-center">
+        <section class="flex flex-col text-xs sm:text-sm md:text-base w-fit justify-center">
+            <nav class="mt-2 sm:mt-3 md:m-10 font-Roboto text-xs">
```
roboto should be the default font, check if you can remove it and it stays the same

kristinatonnessen commented on templates/student-result-teacher-view.html:
```diff
@@ -0,0 +1,196 @@
+
+                <p class="mt-3 text-xs xl:text-lg">Here are the results for <b>$User</b></p>
+
+                <div class="flex flex-row mt-5 font-Roboto">
```
smaller margins for medium screens 
![Skjermbilde 2025-01-31 kl  23 36 18](https://github.com/user-attachments/assets/bab57ab8-047f-4341-b2ad-2e09f16984c4)
![Skjermbilde 2025-01-31 kl  23 38 29](https://github.com/user-attachments/assets/305dceb8-7122-49bc-baf0-d06e02d7e7d1)
![Skjermbilde 2025-01-31 kl  23 38 59](https://github.com/user-attachments/assets/b1839f6d-12a5-4bf7-bc10-eca64afc6e62)

Tomatsuppe11211 committed:
Updated based on feedback

IN this update i have:

1) Removed link to style.css

2) Added rounded borders to navigation (for table) on the bottom of the page.

3) Added the class "secondary-button" to "back to dashboard"-button

4) changed to a pre-set class name for all delete buttons. Scaling have been done too.

5) Font to the page have been changed. Instead of setting fonts multiple times, I have set the body-tag to have font roboto. Some few elements have the font poppins

6) Margin has been reduced between grade and edit button in the table. Scaling almost / just complete

What to work on next time:
- style the navigation for table to make all boxes to look like each other.

Tomatsuppe11211 committed:
update 2

I have changed the sizes on the navigating boxes under the table

kristinatonnessen commented on templates/student-result-teacher-view.html:
```diff
@@ -0,0 +1,195 @@
+
+
+            <div class="flex  justify-center items-center mt-5">
+                <table class="table-auto border-collapse text-left bg-table rounded-lg border-2">
```
there is no border in the design

Tomatsuppe11211 replied:
Forgot to remove before i sent back XD

Border removed

kristinatonnessen commented on templates/student-result-teacher-view.html:
```diff
@@ -0,0 +1,195 @@
+
+
+            <div class="flex  justify-center items-center mt-5">
+                <table class="table-auto border-collapse text-left bg-table rounded-lg border-2">
```
the margins still needs to be removed/made smaller to look like the design, and each line height is a little taller in the design
<img width="208" alt="Skjermbilde 2025-02-01 kl  16 25 02" src="https://github.com/user-attachments/assets/9596dda7-7d53-4166-82d7-0f871b0392ad" />
![Skjermbilde 2025-02-01 kl  16 25 30](https://github.com/user-attachments/assets/fc13fdb1-d643-42cc-b77c-57416bbfcfbe)

Tomatsuppe11211 replied:
Reduced margin and made rows higher

Tomatsuppe11211 committed:
Update 3

Made shortened space bewteen info symbol and grades in the table

gave table rows bigger height

kristinatonnessen commented on templates/student-result-teacher-view.html:
```diff
@@ -50,9 +50,9 @@ <h1 class="text-xs md:text-3xl lg:text-4xl xl:text-5xl font-bold font-Poppins">S
 
```
from around 767px and down the margins are still very big, i would keep all the margins for main in the main or section right below to make it easier to change around as needed
![Skjermbilde 2025-02-01 kl  17 59 48](https://github.com/user-attachments/assets/81998a07-e1ea-4aa4-86bc-d26f548bdeea)
![Skjermbilde 2025-02-01 kl  18 03 01](https://github.com/user-attachments/assets/e82a9ff8-fbf1-4812-b581-65fcfdd90ec1)

Tomatsuppe11211 replied:
You mean margin top?

kristinatonnessen replied:
margin x, and gap + color for the breadcrumbs 
<img width="189" alt="Skjermbilde 2025-02-01 kl  18 57 33" src="https://github.com/user-attachments/assets/179dffb9-4dc1-42e7-aedc-f2d8a8a854e9" />

Tomatsuppe11211 replied:
I changed colors:
![image](https://github.com/user-attachments/assets/a069173e-f6ce-4524-a375-a0715b423c4e)

I have done a little with margin-top and made the search bar wider:
![image](https://github.com/user-attachments/assets/6f2aa11e-7aa9-49f9-bbb6-71c25d546606)

Yet i have problems making the text smaller some places in the file. I think this is still to big:
![image](https://github.com/user-attachments/assets/80748f05-e65b-4fd1-93e2-42645a506f00)

As well i think the margin between grade and info-symbol is way too small.

My coding:
![image](https://github.com/user-attachments/assets/1a204c09-fb6b-483a-9b71-889bc5a1d1c0)

Design file:
![image](https://github.com/user-attachments/assets/219036c1-d4a2-470b-92e4-84b6fee85aa3)

Tomatsuppe11211 committed:
Update 3

I have made margin top smaller, some elements / icons are made to be more fit to the screen.

I can am still available to just adjust font-sizes on some elements but not all.

Tomatsuppe11211 committed:
Update 4

I changed line 28 from "$User" to "$user"

I made the add symbol bigger on line 44 to match figma design more. Still awaiting feedback on the rest.

kristinatonnessen committed:
Merge branch 'page/student_results_task_182' into student-results-teacher.view

Tomatsuppe11211 committed:
update 5

I have added another table for the mobile version. I will work further on making only one of the table visible at a time

kristinatonnessen committed:
Merge branch 'page/student_results_task_182' into student-results-teacher.view

Tomatsuppe11211 committed:
Update 6

I have recreated the tables. FOr now i have left 2 rows for desktop table. I am working on spacing the columns. When that is done, I will add the remaining rows.

Tomatsuppe11211 committed:
Update 7

I have worked on the spacing now. it looks well to me, so i'll be waiting for feedbacks

Tomatsuppe11211 committed:
Removed old HTML coding

Tomatsuppe11211 committed:
Removed final border

Tomatsuppe11211 committed:
Added breakpoints for add result button

myrset reviewed (approved):
Looks good
